nwind-capi/include/nwind_capi.h

```diff
@@ -0,0 +1,84 @@
+/* SPDX-License-Identifier: MIT */
+
+#ifndef NWIND_CAPI_H_
+#define NWIND_CAPI_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Opaque handle for a local address space */
+typedef struct nwind_address_space_ nwind_address_space;
+
+/* Opaque handle for a local unwind context */
+typedef struct nwind_unwind_context_ nwind_unwind_context;
+
+/**
+ * Create a local address space to use for unwinding.
+ *
+ * This type can be reused across threads, as long as the address space stays the same.
+ * When dlopen is called or the address space is otherwise mutated, the address space
+ * needs to be reloaded.
+ *
+ * @param use_shadow_stack set to 1 when you want to use the shadow stack.
+ *
+ * @sa nwind_free_address_space, nwind_reload_address_space
+ * @sa nwind_create_unwind_context, nwind_backtrace
+ */
+nwind_address_space *nwind_create_address_space(int use_shadow_stack);
+
+/**
+ * Free the memory associated with the address space.
+ *
+ * @sa nwind_create_address_space
+ */
+void nwind_free_address_space(nwind_address_space *address_space);
+
+/**
+ * Reload the local address space to re-read the mapping.
+ *
+ * When dlopen is called or the address space is otherwise mutated, the address space
+ * needs to be reloaded.
+ *
+ * @sa nwind_create_address_space
+ */
+nwind_address_space *nwind_reload_address_space(nwind_address_space *address_space);
+
+/**
+ * Create a unwind context to use for unwinding.
+ *
+ * This type can be cached within a single thread and reused across repeated calls
+ * to @c nwind_backtrace.
+ *
+ * @sa nwind_free_unwind_context, nwind_create_address_space, nwind_backtrace
+ */
+nwind_unwind_context *nwind_create_unwind_context(void);
```

Comment:
Should probably include `local` in the name? Which essentially has two meanings: local as in the current thread, and local as in in-process (since the crate also supports unwinding non locally, from another process)